### ISSUE TITLE
Crosstalk between different HTTP requests

### ISSUE THREAD

**Issue by wonderix** [dom96/jester#248]
When running the following code, the last call to the long running request `/read` will also receive all the data from previous (interrupted) calls to`/read` (see assert in line 25). You need the head version of jester to compile this test code.

```nim
import jester
import httpClient
import strutils
import asyncfutures
import asyncdispatch
import nativesockets, httpbeast

let lines = 100
let message = "Hello World\r\n"

proc interruptedRead() {.async} =
    let client = newAsyncHttpClient()
    let future = client.get("http://localhost:5000/read")
    discard await future
    client.close()

proc read() {.async} =
    let client = newAsyncHttpClient()
    let future = client.get("http://localhost:5000/read")
    let response = await future
    let body = response.bodyStream
    var len = 0
    while true:
        let (hasData,msg) = await body.read()
        if not hasData: break
        len += msg.len()
        assert(len <= 5 * lines * message.len())
    client.close()

proc write() {.async} =
    let client = newAsyncHttpClient()
    discard await client.get("http://localhost:5000/write")
    client.close()

proc test() {.async} =
    await sleepAsync(1000)
    for i in 0..10:
        await interruptedRead()
    asyncCheck read()
    asyncCheck write()

asyncCheck test()


var subscriptions: seq[FutureStream[string]]

routes:
  get "/write":
      for s in subscriptions:
        for i in 0..lines:
            await s.write(message)
      resp "OK\r\n"

  get "/read":
    enableRawMode()
    request.sendHeaders(Http200,@({"Content-Type": "text/plain"}))
    let stream = newFutureStream[string]()
    subscriptions.add(stream)
    while true:
        let (hasData, message) = await stream.read()
        if not hasData:
            break
        request.send(message)
    for i in 0..subscriptions.len():
        if subscriptions[i] == stream:
            subscriptions.del(i)
    request.close()
    break route
```

I also saw that data written into one of the `subscriptions` `FuturesStream`s is returned in the call to `/write` (even before the HTTP header!). But I was not able to reproduce this with a simple example.

It seems that the `request` object is somehow shared between different HTTP requests running at the same time ?? Or am I doing something wrong?

**Comment by dom96** [comment]
Is this when running on multiple threads? Can you try without ``--threads:on`` if so?

**Comment by wonderix** [comment]
I'm running it without threads. It's not possible to run it with threads, because the `subscriptions` variable is global.

**Comment by TensorTom** [comment]
Did https://github.com/dom96/httpbeast/pull/35 fix it or do we still need to use the `-d:useStdLib` compile flag?

**Comment by dom96** [comment]
Looks like this is still reproducible, but it indeed doesn't look related to httpbeast (as mentioned here: https://github.com/dom96/httpbeast/pull/35#issuecomment-788174779)

**Comment by dom96** [comment]
Looking into this a bit more it seems this is in fact now fixed.

A) Somehow when I was reproducing it Nim was using a different version of httpbeast (even though I ran `develop`...)
B) I think the reason @marcomq still saw an assert failure is because they didn't change the `5` in the `assert` line, once I did I wasn't able to reproduce this anymore and I assume that is the expected behaviour.

So closing this now. Thanks for reporting and apologies for the long delay. Let me know if this is still reproducible, maybe I misunderstood the code :)
